providers/openshift/provider.go

```diff
@@ -633,6 +633,10 @@ func getKubeAPIURLWithPath(path string) *url.URL {
 	}
 
 	if host := os.Getenv("KUBERNETES_SERVICE_HOST"); len(host) > 0 {
+		// assume IPv6 if host contains colons
+		if strings.IndexByte(host, ':') != -1 {
+			host = "[" + host + "]"
+		}
```

Comment:
a safer check would be to parse this with `net.ParseIP()` and then check if `ip.To4()` succeeds or not

Reply:
or likely even better, wherever this later gets used in a URL, use `net.JoinHostPort()` which will automatically add brackets around an IPv6 address if needed